common/client-core/src/init/types.rs

```diff
@@ -42,24 +39,47 @@ impl SelectedGateway {
     pub fn from_topology_node(
         node: RoutingNode,
         must_use_tls: bool,
+        no_hostname: bool,
```

Comment:
given we have to already weave it through quite a long call stack, how about we also add that `prefer_ipv6` that is hardcoded there and wrap it in some `Config` struct? (because I've got a gut feeling, we'll have to add extra things here soon...)